mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -210,6 +210,7 @@ public void initialize(InputSplit split, TaskAttemptContext newContext) {
       this.context = newContext;
       Table table = ((IcebergSplit) split).table();
       this.io = table.io();
+      this.io.initialize(conf.getValByRegex(""));
```

Comment:
Might worth to check if this regex is faster, or using the iterator interface for Config to create a Map, and adding that to `initialize` is faster

Reply:
I'll actually refactor this to use the `setConf()` method of the `HadoopConfigurable`

Reply:
Also moved the setConf() logic into the table deserialization part in the StorageHandler